tests/Kinepolis/Trailer/YoutubeTrailerRepositoryTest.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CultuurNet\UDB3\Kinepolis\Trailer;
+
+use Broadway\UuidGenerator\UuidGeneratorInterface;
+use CultuurNet\UDB3\Json;
+use CultuurNet\UDB3\Model\ValueObject\MediaObject\Video;
+use CultuurNet\UDB3\Model\ValueObject\Translation\Language;
+use CultuurNet\UDB3\Model\ValueObject\Web\Url;
+use CultuurNet\UDB3\SampleFiles;
+use Google\Service\YouTube\Resource\Search;
+use Google_Service_YouTube;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+
+final class YoutubeTrailerRepositoryTest extends TestCase
+{
+    private TrailerRepository $trailerRepository;
+
+    /**
+     * @var Google_Service_YouTube&MockObject
+     */
+    private $youtubeClient;
+
+    private string $channelId;
+
+    /**
+     * @var UuidGeneratorInterface&MockObject
+     */
+    private $uuidGenerator;
+
+    /**
+     * @var Search&MockObject
+     */
+    private $search;
+
+    public function setUp(): void
+    {
+        $this->search = $this->createMock(Search::class);
+        $this->youtubeClient = $this->createMock(Google_Service_YouTube::class);
+        $this->youtubeClient->search = $this->search;
+        $this->channelId = 'mockChannelId';
+        $this->uuidGenerator = $this->createMock(UuidGeneratorInterface::class);
+
+        $this->trailerRepository = new YoutubeTrailerRepository(
+            $this->youtubeClient,
+            $this->channelId,
+            $this->uuidGenerator,
+            true
+        );
+    }
+
+    /**
+     * @test
+     */
+    public function it_will_only_search_when_enabled(): void
+    {
+        $disabledTrailerRepository = new YoutubeTrailerRepository(
+            $this->youtubeClient,
+            $this->channelId,
+            $this->uuidGenerator,
+            false
+        );
+        $this->search->expects($this->never())->method('listSearch');
+
+        $video = $disabledTrailerRepository->search('Het Smelt');
+        $this->assertNull($video);
+    }
+
+    /**
+     * @test
+     */
+    public function it_will_return_null_if_no_trailers_were_found(): void
+    {
+        $this->search->expects($this->once())->method('listSearch')->with('id,snippet', [
+            'channelId' => $this->channelId,
+            'q' => 'NotFound',
+            'maxResults' => 1,
+        ])->willReturn(
+            Json::decodeAssociatively(
+                SampleFiles::read(__DIR__ . '/../samples/YoutubeNoResults.json')
+            )
+        );
+        $this->uuidGenerator->expects($this->never())->method('generate');
+
+        $video = $this->trailerRepository->search('NotFound');
+        $this->assertNull($video);
```

Comment:
To me it makes more sense, for this to always return a list of items.
If none are found, it returns an empty list

Reply:
The function name `search()` is indeed misleading, but the problem is that Kinepolis almost exclusively only has 1 trailer per movie, so we are never interested in the elements after the first result, because they are the wrong trailers. It either has the correct Trailer as a first searchResult or it doesn't.
Maybe renaming the function to findMatchingTrailer() would be more clear?

Reply:
Another approach could be to throw something like `NoTrailerFound` and have a non-nullable return type.